app/main/views/users.py

```diff
@@ -19,14 +19,25 @@ def auth_user():
     json_payload = json_payload["authUsers"]
     validate_user_auth_json_or_400(json_payload)
 
-    user = User.query.filter(
-        User.email_address == json_payload['emailAddress'].lower()).first()
+    user = User.get_by_email_address(json_payload['emailAddress'].lower())
 
     if user is None:
         return jsonify(authorization=False), 404
-    elif encryption.checkpw(json_payload['password'], user.password):
+    elif encryption.authenticate_user(json_payload['password'], user):
+        user.logged_in_at = datetime.utcnow()
+        user.failed_login_count = 0
+        db.session.add(user)
+        db.session.commit()
+
         return jsonify(users=user.serialize()), 200
     else:
+        login_limit = current_app.config['DM_FAILED_LOGIN_LIMIT']
+        user.failed_login_count += 1
+        if user.failed_login_count >= login_limit:
+            user.locked = True
```

Comment:
I don't think there's a need to set the locked flag when we can check the counter directly.
Is there any other use for the `user.locked` flag? And how is it different from `user.active`?

Can we check if they're set for any users in the user DB dump and decide whether we need them at all?
E.g. if the `locked` is not set for any users, I'd replace it with a property that'd check the `failed_login_count` value.
If it is set, then we should probably find out why and avoid overwriting the flag value so that we don't unlock a user later thinking they were locked for the failed attempts when the real reason might've been different.